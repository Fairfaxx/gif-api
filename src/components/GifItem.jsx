const GifItem = ({ title, url, id }) => {
  return (
    <div className="card" key={id}>
      <h4>{title}</h4>
      <img src={url} alt={title} />
    </div>
  );
};

export default GifItem;
