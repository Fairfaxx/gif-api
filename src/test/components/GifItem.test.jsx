import { render, screen } from '@testing-library/react';
import GifItem from '../../components/GifItem';

describe('Test GifItem component', () => {
  const title = 'test title';
  const url = 'https://dragon-ball.com/goku.jpg';
  const alt = 'test alt';
  test('should match snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should display the image with the url and alt correctly', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should display the title', () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
