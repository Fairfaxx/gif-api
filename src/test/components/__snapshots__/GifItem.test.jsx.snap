// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem component should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h4>
      test title
    </h4>
    <img
      alt="test title"
      src="https://dragon-ball.com/goku.jpg"
    />
  </div>
</div>
`;
